SimG4CMS/PPS/src/PPSPixelSD.cc

```diff
@@ -0,0 +1,437 @@
+
+  ThetaAtEntry = aStep->GetPreStepPoint()->GetPosition().theta();
+  PhiAtEntry = aStep->GetPreStepPoint()->GetPosition().phi();
+
+  ParentId = theTrack->GetParentID();
+  Vx = theTrack->GetVertexPosition().x();
+  Vy = theTrack->GetVertexPosition().y();
+  Vz = theTrack->GetVertexPosition().z();
+}
+
+bool PPSPixelSD::HitExists() {
+  if (primaryID < 1) {
+    edm::LogWarning("PPSSim") << "***** PPSPixelSD error: primaryID = " << primaryID << " maybe detector name changed";
+  }
+
+  // Update if in the same detector, time-slice and for same track
+  if (tSliceID == tsID && unitID == previousUnitID) {
+    UpdateHit();
+    return true;
+  }
+
+  // Reset entry point for new primary
+  if (primaryID != primID)
+    ResetForNewPrimary();
+
+  //look in the HitContainer whether a hit with the same primID, unitID,
+  //tSliceID already exists:
+  bool found = false;
+
+  for (int j = 0; j < theHC->entries() && !found; j++) {
+    PPSPixelG4Hit* aPreviousHit = (*theHC)[j];
+    if (aPreviousHit->getTrackID() == primaryID && aPreviousHit->getTimeSliceID() == tSliceID &&
+        aPreviousHit->getUnitID() == unitID) {
+      currentHit = aPreviousHit;
+      found = true;
+    }
+  }
+
+  if (found) {
+    UpdateHit();
+    return true;
+  }
+  return false;
+}
+
+void PPSPixelSD::CreateNewHit() {
+#ifdef debug
+  LogDebug("PPSSim") << "PPSPixelSD CreateNewHit for"
+                     << " PV " << currentPV->GetName() << " PVid = " << currentPV->GetCopyNo()
+                     << " MVid = " << currentPV->GetMother()->GetCopyNo() << " Unit " << unitID << "\n"
+                     << " primary " << primaryID << " time slice " << tSliceID << " For Track  "
+                     << theTrack->GetTrackID() << " which is a " << theTrack->GetDefinition()->GetParticleName();
+
+  if (theTrack->GetTrackID() == 1) {
+    LogDebug("PPSSim") << " of energy " << theTrack->GetTotalEnergy();
+  } else {
+    LogDebug("PPSSim") << " daughter of part. " << theTrack->GetParentID();
+  }
+
+  if (theTrack->GetCreatorProcess() != NULL)
+    LogDebug("PPSSim") << theTrack->GetCreatorProcess()->GetProcessName();
+  else
+    LogDebug("PPSSim") << "NO process";
+#endif
+
+  currentHit = new PPSPixelG4Hit;
+  currentHit->setTrackID(primaryID);
+  currentHit->setTimeSlice(tSlice);
+  currentHit->setUnitID(unitID);
+  currentHit->setIncidentEnergy(incidentEnergy);
+
+  currentHit->setPabs(Pabs);
+  currentHit->setTof(Tof);
+  currentHit->setEnergyLoss(Eloss);
+  currentHit->setParticleType(ParticleType);
+  currentHit->setThetaAtEntry(ThetaAtEntry);
+  currentHit->setPhiAtEntry(PhiAtEntry);
+
+  currentHit->setMeanPosition(Posizio);
+  currentHit->setEntryPoint(theEntryPoint);
+  currentHit->setExitPoint(theExitPoint);
+
+  currentHit->setParentId(ParentId);
+  currentHit->setVx(Vx);
+  currentHit->setVy(Vy);
+  currentHit->setVz(Vz);
+
+  UpdateHit();
+
+  StoreHit(currentHit);
+}
+
+void PPSPixelSD::CreateNewHitEvo() {
+  currentHit = new PPSPixelG4Hit;
```

Comment:
should be a unique_ptr constructed with make_unique

Reply:
Kept, as for other G4Hit objects.